components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -65,6 +65,8 @@ open class GleanInternalAPI internal constructor () {
 
     internal lateinit var pingStorageEngine: PingStorageEngine
 
+    internal val tasksQueuedBeforeInit: MutableList<() -> Unit> = mutableListOf()
```

Comment:
I wonder if there's any lock-free data structure, in Kotlin, to provide thread-safe access to this.

Reply:
[ConcurrentLinkedQueue](https://developer.android.com/reference/java/util/concurrent/ConcurrentLinkedQueue.html) is probably the best bet